aircan/dependencies/google_cloud/bigquery_handler_v2.py

```
@@ -0,0 +1,152 @@
+
+            # Join the list of lines into a single string with newlines.
+            error_report_string = "\n".join(error_lines)
+            logging.info(error_report_string)
+            status_dict = {
+                'res_id': ckan_conf.get('resource_id'),
+                'state': 'failed',
+                'message': error_report_string,
+                'dag_run_id': ckan_conf.get('dag_run_id')
+            }
+            aircan_status_update(ckan_conf.get('site_url'), ckan_conf.get('api_key'), status_dict)
+            raise AirflowCKANException('Data ingestion has failed.', str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)